mdc_weight_mgmt_xml_import/models/maintenance_equipment.py

```diff
@@ -0,0 +1,151 @@
+import traceback
+import logging
+import re
+
+
+class MaintenanceEquipment(models.Model):
+    _inherit = 'maintenance.equipment'
+
+    mdc_xml_weight_in_path = fields.Char(
+        string="XML Input Folder Path",
+        help="Path to the folder where unprocessed XML files are placed."
+    )
+    mdc_xml_weight_processed_path = fields.Char(
+        string="XML Processed Folder Path",
+        help="Path to the folder where processed XML files are moved."
+    )
+
+    @api.model
+    def _cron_import_xml_weight(self):
+        logger = logging.getLogger(__name__)
+        errors = []
+        for record in self.env['maintenance.equipment'].search([
+                    ('mdc_xml_weight_in_path', '!=', False),
+                    ('mdc_xml_weight_processed_path', '!=', False),
+                ]):
+            in_path = Path(record.mdc_xml_weight_in_path)
+            processed_path = Path(record.mdc_xml_weight_processed_path)
+
+            try:
+                in_path.mkdir(parents=True, exist_ok=True)
+                processed_path.mkdir(parents=True, exist_ok=True)
+            except Exception as e:
+                logger.error("Can't create folders: %s", e)
+                errors.append({
+                    'equipment_id': record.id,
+                    'file_name': None,
+                    'error_message': (_("Folder creation failed: %s",e)),
+                    'traceback': traceback.format_exc(),
+                })
+                continue
+
+            try:
+                files = [f for f in in_path.iterdir() if f.is_file()]
+            except Exception as e:
+                logger.error("Error listing files: %s", e)
+                errors.append({
+                    'equipment_id': record.id,
+                    'file_name': None,
+                    'error_message': (_("Listing files failed: %s",e)),
+                    'traceback': traceback.format_exc(),
+                })
+                continue
+
+            if not files:
+                logger.info("No files to process in %s", in_path)
+                continue
+
+            for file in files:
+                if not file.name.lower().endswith('.xml'):
+                    logger.warning("Skipping non-XML file: %s", file.name)
+                    continue
+                try:
+                    self._process_xml_file(file, processed_path, record)
+                except Exception as e:
+                    logger.error("Error processing file %s: %s", file.name, e)
+                    errors.append({
+                        'equipment_id': record.id,
+                        'file_name': file.name,
+                        'error_message': (_("Processing failed: %s",e)),
+                        'traceback': traceback.format_exc(),
+                    })
+
+        if errors:
+            error_model = self.env['mdc.weight.record.error'].sudo()
+            created_errors = error_model.browse([error_model.create(e).id for e in errors])
```

Comment:
Why not using create multi capabilities here?
```suggestion
            created_errors = error_model.create(errors)
```